hudi-common/src/main/java/org/apache/hudi/common/util/JsonUtils.java

```diff
@@ -44,6 +44,8 @@ public class JsonUtils {
     MAPPER.setVisibility(PropertyAccessor.IS_GETTER, JsonAutoDetect.Visibility.NONE);
     MAPPER.setVisibility(PropertyAccessor.SETTER, JsonAutoDetect.Visibility.NONE);
     MAPPER.setVisibility(PropertyAccessor.CREATOR, JsonAutoDetect.Visibility.NONE);
+
+    registerModules();
```

Comment:
@yihua 
It was because of the use of JsonUtils in [MetadataConversionUtils.convertCommitMetadata()](https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/MetadataConversionUtils.java#L257), not on the adapter side.
I think JsonUtils needs it as a defense logic because there is no application according to the spark version.

Reply:
Sg